mono/metadata/profiler-events.h

```diff
@@ -82,7 +82,7 @@ MONO_PROFILER_EVENT_1(method_end_invoke, MethodEndInvoke, MonoMethod *, method)
 MONO_PROFILER_EVENT_1(exception_throw, ExceptionThrow, MonoObject *, exception)
 MONO_PROFILER_EVENT_4(exception_clause, ExceptionClause, MonoMethod *, method, uint32_t, index, MonoExceptionEnum, type, MonoObject *, exception)
 
-MONO_PROFILER_EVENT_2(gc_event, GCEvent, MonoProfilerGCEvent, event, uint32_t, generation)
+MONO_PROFILER_EVENT_3(gc_event, GCEvent, MonoProfilerGCEvent, event, uint32_t, generation, uint32_t, serial)
```

Comment:
Two things:

1. This is public API, so we cannot change the signature. You have to introduce a new callback and make the runtime invoke both the old and new one.
2. The `serial` parameter should be `mono_bool`.